File: Version_1/src/environment/environment.py
"""Top-level orchestrator tying room state to the rest of the game (mirrors
Version_0's Environment, trimmed to what Phase 1 needs -- no reward/step
loop yet, just room state)."""

from environment.state import RoomState


class Environment:
    def __init__(self):
        self.state = RoomState()

    def reset(self):
        self.state = RoomState()
        return self.state

File: Version_2/src/environment/environment.js
/**
 * Top-level orchestrator tying room state to the rest of the game. Direct
 * port of Version_1/src/environment/environment.py.
 */
window.Game = window.Game || {};

Game.Environment = (function () {
  class Environment {
    constructor() {
      this.state = new Game.State.RoomState();
    }

    reset() {
      this.state = new Game.State.RoomState();
      return this.state;
    }
  }

  return { Environment };
})();
